src/typed-method-types/workflows/triggers/webhook_test.ts

```diff
@@ -0,0 +1,40 @@
+import { assertEquals } from "https://deno.land/std@0.99.0/testing/asserts.ts";
+import { WebhookTrigger } from "./webhook.ts";
+import { TriggerTypes } from "./mod.ts";
+
+Deno.test("Webhook Triggers can set the type using the string", () => {
+  const webhook: WebhookTrigger = {
+    type: "webhook",
+    name: "test",
+    workflow: "example",
+    inputs: {},
+  };
+  assertEquals(webhook.type, TriggerTypes.Webhook);
+});
+
+Deno.test("Webhook Triggers can set the type using the TriggerTypes object", () => {
+  const webhook: WebhookTrigger = {
+    type: TriggerTypes.Webhook,
+    name: "test",
+    workflow: "example",
+    inputs: {},
+  };
+  assertEquals(webhook.type, TriggerTypes.Webhook);
+});
+
+Deno.test("Webhook Trigger Filters support an optional webhook object", () => {
+  const webhook: WebhookTrigger = {
+    type: TriggerTypes.Webhook,
+    name: "test",
+    workflow: "example",
+    inputs: {},
+    webhook: {
+      filter: {
+        root: {
+          statement: "1 === 1",
```

Comment:
In the olden days we tried to do some fancy things with the dynamic inputs. For example if the trigger was an reaction event trigger, you'd get an autocomplete for the fields which we knew were part of that event (e.g. `reaction_name` or whatever). Are those types in the SDK anymore? If not, do we plan to add them back or are we going to go ahead without them?

Either way, what would declaring a trigger with a filter look like if we wanted to include a variable binding like `statement: "{{data.reaction}} === 'panda'"`?

Reply:
Yes! All this typing looks pretty much the same as it used to, but the API library that we're in here has no context from the trigger so during this initial phase devs will need to know what they're doing.

Eventually we plan to reintroduce `DefineTrigger` to the SDK library and that will include support for filters (the old `withFilter` method).